tests/fixture/tmpnet/start_kind_cluster.go

```diff
@@ -8,49 +8,137 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
+	"strings"
 
 	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
 
 	"github.com/ava-labs/avalanchego/utils/logging"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// TODO(marun) This should be configurable
+	DefaultTmpnetNamespace = "tmpnet"
+
+	KindKubeconfigContext = "kind-kind"
+
+	// TODO(marun) Check for the presence of the context rather than string matching on this error
+	missingContextMsg = `context "` + KindKubeconfigContext + `" does not exist`
 )
 
-// CheckClusterRunning checks if the configured cluster is accessible.
-// TODO(marun) Maybe differentiate between configuration and endpoint errors
-func CheckClusterRunning(log logging.Logger, configPath string, configContext string) error {
+// StartKindCluster starts a new kind cluster with integrated registry if one is not already running.
+func StartKindCluster(ctx context.Context, log logging.Logger, configPath string) error {
+	configContext := KindKubeconfigContext
+
+	clusterRunning, err := isKindClusterRunning(log, configPath)
+	if err != nil {
+		return err
+	}
+	if clusterRunning {
+		log.Info("local kind cluster already running",
+			zap.String("kubeconfig", configPath),
+			zap.String("kubeconfigContext", configContext),
+		)
+	} else {
+		log.Info("attempting to start local kind cluster",
+			zap.String("kubeconfig", configPath),
+			zap.String("kubeconfigContext", configContext),
+		)
+
+		startCtx, cancel := context.WithTimeout(ctx, DefaultNetworkTimeout)
+		defer cancel()
+		cmd := exec.CommandContext(startCtx, "kind-with-registry.sh")
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("failed to run kind-with-registry.sh: %w", err)
+		}
+	}
+
 	clientset, err := GetClientset(log, configPath, configContext)
 	if err != nil {
 		return err
 	}
-	// Check if the configured context can reach a cluster endpoint
-	_, err = clientset.Discovery().ServerVersion()
-	return err
+
+	return ensureNamespace(ctx, log, clientset, DefaultTmpnetNamespace)
 }
 
-// StartKindCluster starts a new kind cluster if one is not already running.
-func StartKindCluster(ctx context.Context, log logging.Logger, configPath string, configContext string) error {
-	err := CheckClusterRunning(log, configPath, configContext)
-	if err == nil {
-		log.Info("kubernetes cluster already running",
+// isKindClusterRunning determines if a kind cluster is running with context kind-kind
+func isKindClusterRunning(log logging.Logger, configPath string) (bool, error) {
+	configContext := KindKubeconfigContext
+
+	_, err := os.Stat(configPath)
+	if os.IsNotExist(err) {
```

Comment:
We aren't consistent here, but these error checks are documented as being less preferred rather than `errors.Is`:

> This function predates `errors.Is`. It only supports errors returned by the os package. New code should use `errors.Is(err, fs.ErrNotExist)`.